### PR TITLE
User and Review CRUD Features and Tests

### PR DESCRIPTION
**Pull request by Dodie324**

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  def update
+    @sitter = User.find(params[:sitter_id])
+    @review = Review.find(params[:id])
+
+    if @review.update(review_params)
+      flash[:notice] = "Review updated!"
+      redirect_to sitter_path(@sitter)
+    else
+      @errors = @review.errors.full_messages
```
Comment (omidbachari) [inline app/controllers/reviews_controller.rb]: No need to create an assignment here in the controller. Just call `@review.errors.full_messages` in the view and display accordingly.

File: app/views/reviews/_index.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<div class="rating">
+  <% @reviews.each do |review| %>
+    <%= review.rating %>
+    <%= review.body %>
+    <% if current_user %>
+      <li><%= link_to "Edit Review", edit_sitter_review_path(@sitter, review)  %></li>
+    <% end %>
+    <% if current_user %>
```
Comment (omidbachari) [inline app/views/reviews/_index.html.erb]: Is there a reason why you needed two conditionals? Just test the existence of `current_user` once.

File: spec/features/users/user_signs_up_spec.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  # Acceptance Criteria:
+  # * I must specify a valid email address,
+  #   password, and password confirmation
+  # * If I don't specify the required information, I am presented with
+  #   an error message
+
+  scenario 'provide valid registration information as regular user' do
+    visit new_user_registration_path
+
+    fill_in 'Email', with: 'john@example.com'
```
Comment (omidbachari) [inline spec/features/users/user_signs_up_spec.rb]: Not wise to key in lots of strings here. Better to create an object and call on its attributes.

**Comment by omidbachari** [comment]
Resolve feedback and merge

File: app/views/reviews/_index.html.erb (hunk 2 above)
Comment (Dodie324) [inline app/views/reviews/_index.html.erb]: This was a copy and paste error. Deleted the additional conditional. Good eye.

**Pull request merged**